### PR TITLE
update guava and appengine-plugins-core

### PR DESCRIPTION
**Pull request by elharo**
Not quite the latest but this does cure https://www.google.com/url?q=https://cve.mitre.org/cgi-bin/cvename.cgi?name%3DCVE-2018-10237&sa=D&usg=AFQjCNHP2RVzxg0gz-WpL_Kqsxb6q4-Nqw

### CODE REVIEW

**Comment by chanseokoh** [comment]
Updating Guava needs a lot of files to be touched, including fixing tests. Try `$ git grep '21\.0\.0'` and see what you get.

BTW, I always have no confidence with Guava versions. Honestly, I still don't get the reason we have the `;version="[21.0.0,22.0.0)`  directives(?).

**Comment by elharo** [comment]
I think this is going to need a matching release of appengine-plugins-core first.

**Comment by chanseokoh** [comment]
21.0.0 still in `category.xml`. This should also be fixed. But I cannot vouch that this will make things work.

```
$ git grep '21\.0\.0'
eclipse/ide-target-platform/category.xml:   <bundle id="com.google.guava" version="21.0.0"/>
plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java:        "Import-Package", "com.google.common.", "version=\"[21.0.0,22.0.0)\"");
plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java:        "Require-Bundle", "com.google.guava", "bundle-version=\"[21.0.0,22.0.0)\"");
```

**Comment by chanseokoh** [comment]
For `category.xml`, I think it specifies bundle versions.

**Comment by briandealwis** [comment]
> Honestly, I still don't get the reason we have the ;version="[21.0.0,22.0.0) directives(?).

It was to prevent us from being bound to earlier and later versions of Guava, which would lead to package-uses errors.  IIRC Eclipse usually ships with older Guava (getting everybody to update in tandem is hard), and AERI was using a newer Guava.  Locking in the versions brought us sanity.

Maybe we should just use Require-Bundle.

**Comment by elharo** [comment]
Having trouble building the target platform locally. It's still picking up appengine-plugins-core 0.6.3.

**Comment by elharo** [comment]
Now seeing a lot of errors due to missing data binding types:

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.1.0:compile (default-compile) on project com.google.cloud.tools.eclipse.appengine.deploy.ui: Compilation failure: Compilation failure: 
[ERROR] /Users/elharo/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java:[190] 
[ERROR] 	Binding binding = bindingContext.bindValue(accountSelectorObservableValue, accountEmailModel,
[ERROR] 	                                 ^^^^^^^^^
[ERROR] The method bindValue(IObservableValue, IObservableValue, UpdateValueStrategy, UpdateValueStrategy) in the type DataBindingContext is not applicable for the arguments (AccountSelectorObservableValue, IObservableValue<String>, UpdateValueStrategy, UpdateValueStrategy)
[ERROR] /Users/elharo/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java:[331] 
[ERROR] 	model.save();
[ERROR] 	      ^^^^
[ERROR] The method save() from the type DeployPreferences refers to the missing type BackingStoreException
[ERROR] /Users/elharo/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java:[352] 
[ERROR] 	accountSelector = new AccountSelector(this, loginService);
[ERROR] 	                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The constructor AccountSelector(Composite, IGoogleLoginService) refers to the missing type Composite
[ERROR] /Users/elharo/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java:[355] 
[ERROR] 	accountSelector.setLayoutData(accountSelectorGridData);
[ERROR] 	                ^^^^^^^^^^^^^
[ERROR] The method setLayoutData(GridData) is undefined for the type AccountSelector
[ERROR] /Users/elharo/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java:[407] 
[ERROR] 	projectSelector.addSelectionChangedListener(
[ERROR] 	                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The method addSelectionChangedListener(ISelectionChangedListener) in the type ProjectSelector is not applicable for the arguments (ProjectSelectorSelectionChangedListener)
[ERROR] /Users/elharo/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java:[500] 
[ERROR] 	latestGcpProjectQueryJob = new GcpProjectQueryJob(selectedCredential,
[ERROR]           projectRepository, projectSelector, bindingContext, isLatestQueryJob);
[ERROR] 	                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The constructor GcpProjectQueryJob(Credential, ProjectRepository, ProjectSelector, DataBindingContext, Predicate<Job>) refers to the missing type DataBindingContext
[ERROR] /Users/elharo/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java:[500] 
[ERROR] 	latestGcpProjectQueryJob = new GcpProjectQueryJob(selectedCredential,
[ERROR]           projectRepository, projectSelector, bindingContext, isLatestQueryJob);
[ERROR] 	                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Type mismatch: cannot convert from GcpProjectQueryJob to Job
[ERROR] /Users/elharo/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java:[618] 
[ERROR] 	String selectedEmail = accountSelectorObservableValue.getValue();
[ERROR] 	                                                      ^^^^^^^^
[ERROR] The method getValue() is undefined for the type AccountSelectorObservableValue
[ERROR] 22 problems (8 errors, 14 warnings)
```

**Comment by chanseokoh** [comment]
Haven't done any manual plugin test, but code-wise, looks good to me.
```
$ git grep '21\.0\.0'
$ git grep '22\.0\.0'
$ git grep '0\.6\.3'
$ git grep '0\.6\.6'
(shows matching lines)
```
For me, `$ mvn verify` works locally.

**Comment by chanseokoh** [comment]
But when I try to update the target platform with `$ (cd eclipse ; mvn -Pide-target-platform package)`, it still packages up appengine-plugins-core 0.6.3 instead of 0.6.6. Not sure what's going on.

**Comment by chanseokoh** [comment]
> But when I try to update the target platform with $ (cd eclipse ; mvn -Pide-target-platform package), it still packages up appengine-plugins-core 0.6.3 instead of 0.6.6. Not sure what's going on.

Looks like Eclipse is somehow caching the info about software sites (in our case, ours is a local filesystem software site), and I needed to tell Eclipse to explicitly reload the existing site. Can be done through Windows > Preferences > Plug-in Development > Target Platform > select your target platform > Edit... > select the item > Edit... > Manage..., and there, you will see your local target platform site "ide-target-platform.repo (file:/usr/local/...)" > select it and click "Reload".

File: plugins/com.google.cloud.tools.eclipse.ui.status/OSGI-INF/polling.xml [hunk 1]
```diff
@@ -3,6 +3,6 @@
    <service>
       <provide interface="com.google.cloud.tools.eclipse.ui.status.GcpStatusMonitoringService"/>
    </service>
-   <reference bind="setProxyService" cardinality="0..1" interface="org.eclipse.core.net.proxy.IProxyService" name="ProxyService" policy="dynamic" unbind="unsetProxyService"/>
+   <reference bind="setProxyService" cardinality="0..1" interface="IProxyService" name="ProxyService" policy="dynamic" unbind="unsetProxyService"/>
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.ui.status/OSGI-INF/polling.xml]: Should we revert this? My Eclipse restore this to `org.eclipse.core.net.proxy.IProxyService` automatically.

File: plugins/com.google.cloud.tools.eclipse.googleapis/OSGI-INF/com.google.cloud.tools.eclipse.googleapis.internal.GoogleApiFactory.xml [hunk 2]
```diff
@@ -3,6 +3,6 @@
    <service>
       <provide interface="com.google.cloud.tools.eclipse.googleapis.IGoogleApiFactory"/>
    </service>
-   <reference bind="setProxyService" cardinality="0..1" interface="org.eclipse.core.net.proxy.IProxyService" name="ProxyService" policy="dynamic" unbind="unsetProxyService"/>
+   <reference bind="setProxyService" cardinality="0..1" interface="IProxyService" name="ProxyService" policy="dynamic" unbind="unsetProxyService"/>
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.googleapis/OSGI-INF/com.google.cloud.tools.eclipse.googleapis.internal.GoogleApiFactory.xml]: Should we revert this? My Eclipse restore this to `org.eclipse.core.net.proxy.IProxyService` automatically.

File: plugins/com.google.cloud.tools.eclipse.ui.status/OSGI-INF/polling.xml (hunk 1 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.ui.status/OSGI-INF/polling.xml]: And my Eclipse moved it. I don't know why, either way.

File: plugins/com.google.cloud.tools.eclipse.googleapis/OSGI-INF/com.google.cloud.tools.eclipse.googleapis.internal.GoogleApiFactory.xml (hunk 2 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.googleapis/OSGI-INF/com.google.cloud.tools.eclipse.googleapis.internal.GoogleApiFactory.xml]: Should be reverted: the interface is a fully-qualified type name.

File: plugins/com.google.cloud.tools.eclipse.ui.status/OSGI-INF/polling.xml (hunk 1 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.ui.status/OSGI-INF/polling.xml]: Should be reverted: the interface is a fully-qualified type name.

**Comment by briandealwis** [comment]
> But when I try to update the target platform with $ (cd eclipse ; mvn -Pide-target-platform package), it still packages up appengine-plugins-core 0.6.3 instead of 0.6.6. Not sure what's going on.

Works for me.

File: plugins/com.google.cloud.tools.eclipse.googleapis/OSGI-INF/com.google.cloud.tools.eclipse.googleapis.internal.GoogleApiFactory.xml (hunk 2 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.googleapis/OSGI-INF/com.google.cloud.tools.eclipse.googleapis.internal.GoogleApiFactory.xml]: will revert, though why this happened I don't know. I didn't touch this file. Eclipse just decided it needed to be other than it was. :-(

**Pull request merged**